webapp/components/Map/MapOptions.js

```diff
@@ -36,6 +36,24 @@ export const MapOptions = () => {
                 onChange={(value) => onOptionUpdate({ option: 'showMarkersLabels', value })}
               />
             </FormItem>
+            <FormItem label={i18n.t('mapView.options.showSamplingPolygon')}>
+              <Checkbox
+                checked={options.showSamplingPolygon}
+                onChange={(value) => onOptionUpdate({ option: 'showSamplingPolygon', value })}
+              />
```

Comment:
you can use currification to avoid re-renders

const a = (x) => (value) => doSmthg with x and value

<component onE={a(xValue)} />

Reply:
True, just did like that since it is not a long function so it does not affect performance. Will change.

Reply:
Also, I just literally followed the convention there already was, the existing code was like that and I just added free more